addons/auth_totp/views/user_preferences.xml

```diff
@@ -1,51 +1,64 @@
-            <tree>
-                <field name="totp_enabled"/>
-            </tree>
+            <xpath expr="//search" position="inside">
+                <separator/>
+                <filter name="totp_enabled" string="Two-factor authentication Enabled" domain="[('totp_secret','!=',False)]"/>
+                <separator/>
+                <filter name="totp_disabled" string="Two-factor authentication Disabled" domain="[('totp_secret','=',False)]"/>
+            </xpath>
         </field>
     </record>
+
     <record model="ir.ui.view" id="view_totp_form">
         <field name="name">user form: add totp status</field>
         <field name="model">res.users</field>
         <field name="inherit_id" ref="base.view_users_form"/>
         <field name="arch" type="xml">
-            <xpath expr="//page[@name='references']/group[1]" position="before">
-                <field name="totp_enabled" invisible="1"/>
-                <group attrs="{'invisible': [('totp_enabled', '!=', False)]}">
-                    <div>
-                        <span class="alert alert-info" role="status">
-                            <i class="fa fa-warning"/>
-                            Two-factor authentication not enabled
-                        </span>
-                    </div>
-                </group>
-                <group attrs="{'invisible': [('totp_enabled', '=', False)]}">
-                    <div>
-                        <span class="text-success">
-                            <i class="fa fa-check-circle"/>
-                            Two-factor authentication enabled
-                        </span>
-                    </div>
-                </group>
+            <xpath expr="//page[@name='preferences']" position="after">
+                <page string="Account Security" name="security" attrs="{'invisible': [('id', '=', False)]}">
+                    <field name="totp_enabled" invisible="1"/>
+                    <!-- For own user, allow to activate the two-factor Authentication -->
+                    <group>
+                        <div>
+                            <div class="o_horizontal_separator d-flex align-items-center mt-0">Two-factor Authentication
+                                <div attrs="{'invisible': [('totp_enabled', '!=', False)]}">
+                                    <button attrs="{'invisible': &quot;[('id', '=', uid)]&quot;}" name="totp_enable_wizard"
+                                            disabled="1" type="object" class="fa fa-toggle-off o_auth_2fa_btn disabled" aria-label="Enable 2FA"></button>
+                                    <button attrs="{'invisible': &quot;[('id', '!=', uid)]&quot;}" name="totp_enable_wizard"
+                                        type="object" class="fa fa-toggle-off o_auth_2fa_btn disabled" aria-label="Enable 2FA"></button>
+                                    <button groups="base.group_erp_manager" attrs="{'invisible': &quot;[('id', '=', uid)]&quot;}"
```

Comment:
Are quotes necessary ? uid is not in eval context ?

Reply:
without quot, kaboom